types/algoliasearch/index.d.ts

```diff
@@ -1880,9 +1881,254 @@ declare namespace algoliasearch {
+
+        /**
+         * Reverse geocoding means converting a location (latitude and longitude) to a readable address.
+         * https://community.algolia.com/places/api-clients.html#endpoints
+         */
+        reverse(e: QueryReverseInterface): Promise<ResultSearchInterface>;
+    }
+
+    /**
+     * Params for endpoint reverse.
+     * https://community.algolia.com/places/api-clients.html#reverse-geocoding
+     */
+    interface QueryReverseInterface {
+        /**
+         * Force to first search around a specific latitude longitude.
+         * The option value must be provided as a string: latitude,longitude like 12.232,23.1.
+         * The default is to search around the location of the user determined via his IP address (geoip).
+         * https://community.algolia.com/places/api-clients.html#api-options-aroundLatLng
+         */
+        aroundLatLng: string;
+        /**
+         * Restrict the search results to a specific type.
+         * https://community.algolia.com/places/api-clients.html#api-options-type
+         */
+        hitsPerPage?: number;
+        /**
+         * If specified, restrict the search results to a single language. You can pass two letters country codes (ISO 639-1).
+         * Warning: language parameter is case sensitive and should be lowercase otherwise it will fallback to default language.
+         * https://community.algolia.com/places/api-clients.html#api-options-language
+         */
+        language?: string;
+    }
+
+
+    /**
+     * Result of search.
+     * https://community.algolia.com/places/api-clients.html#json-answer
+     */
+    interface ResultSearchInterface {
+        /**
+         * Contains all the hits matching the query.
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        hits: HitInterface[];
+        /**
+         * Query fallback if query retrieve any result
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        degradedQuery: boolean;
+        /**
+         * Number of total hits matching the query.
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        nbHits: number;
+        /**
+         * GET parameters used to perform the search.
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        params: string;
+        /**
+         * Engine processing time (excluding network transfer).
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        processingTimeMS: number;
+        /**
+         * Query used to perform the search.
+         * https://community.algolia.com/places/api-clients.html#json-answer
+         */
+        query: string;
+    }
+
+
+    /**
+     * Hit of search.
+     * https://community.algolia.com/places/api-clients.html#api-suggestion-name
+     */
+    interface HitInterface {
```

Comment:
the HitInterface shape is dependent on whether the language parameter was provided or not.

If a supported language was provided as a query parameter the interface looks like this:
```ts
interface LocalizedHitInterface {
  administrative?: string[];
  country: string;
  country_code: string;
  county?: string[];
  city?: string[];
  postcodes?: string[];
  population?: number;
  geoloc: ...
  
}